File: Journey/028/projects/001/YouGuess.py
import random

def adivina():
    print('Dame un rango de números... ')
    rangoInf = int(input('Dime el número inferior: '))
    rangoSup = int(input('Dime el número superior: '))
    numeroAl = random.randrange (rangoInf, rangoSup)
    
    def comparaNumero(numeroAl):
        tunumero = int(input('Dime un número: '))

        if tunumero < numeroAl:
            print ('Ups! El número aleatorio es mayor a tu número')

        elif tunumero == numeroAl:
            print ('Perfect! El número aleatorio es igual a tu número')
            print ('End of Game')

            play = input('Play again (y/n)? ')
            if play == 'y':
                adivina()
            else:
                return False
        else:
            print ('Ups! El número aleatorio es menor a tu número')

    for _ in range (15):
        loop = True
        loop = comparaNumero(numeroAl)
        if loop == False:
            break
    
adivina()

File: Journey/028/projects/001/YouGuess.R
youGuess <- function() {
  print('Give me a range of numbers...')
  rangeInf <- as.integer(readline(prompt='Inferior number: '))
  rangeSup <- as.integer(readline(prompt='Superior number: '))
  compuNum <- sample(rangeInf:rangeSup, 1)  # '1' is the number of samples
  # print(compuNum)
  compareNum <- function(compuNum) {
    yourNum <- as.integer(readline(prompt='Give me a number: '))
    
    if (yourNum < compuNum) {
      print ('Ups! Computer number is greater than your number')
      
    } else if (yourNum == compuNum) {
      print ('Perfect! Computer number is the same as your number')
      print ('End of Game')
      playAgain <- readline(prompt='Play again? (y/n): ')
      if (playAgain == 'y'){
        youGuess()
      } else {
        return (FALSE)
      }
      
    } else {
        print ('Ups! Computer number is lower than your number')
    }
  }
  for (i in 1:10) {
    loop = TRUE
    loop = compareNum(compuNum)
    if (loop == FALSE) {
      print('Thank you, Bye!')
      break
    }
  }
  if (loop != FALSE) {
    print('Sorry, no more opportunities')
  }
}

youGuess()
